docs: Update FAQ and addon author guide to mention testing strategies for non-NPM-published branches / forks (#1734)
embroider-build/embroider: docs/porting-addons-to-v2.md

      "version": 2
    }
    ```

> 16. Now we are ready to build the addon. In the `addon` directory, run `yarn start`. This will compile your authored code into the `dist/` folder in v2 format.

Reviewer: ```suggestion
16. Now we are ready to build the addon. In the `addon` directory, run `npm start`. This will compile your authored code into the `dist/` folder in v2 format.
```

I see that this document previously mentions yarn.

We should probably remove that altogether, as yarn@v1 often has a rough time with correctly resolving dependencies :thinking:

Reviewer: oh ok, there is a lil disclaimer here at the top, nevermind

> Our recommended way to manage these multiple packages is using a monorepo, via Yarn, NPM, or PNPM Workspaces. The example in this guide assumes a Yarn Workspaces monorepo.

## Part 5: Testing your addon

> In a separate shell, you should be able to go into the `test-app` directory and run `yarn start` or `yarn test` and see that your tests are passing.

Reviewer: ```suggestion
In a separate shell, you should be able to go into the `test-app` directory and run `npm start` or `npm test` and see that your tests are passing.
```

> Note that v2 addon needs to be built, so you can't easily point package.json files at the git repo of your addon for easy cross-repo testing before it is published to NPM. There are [workflows you can use](https://github.com/embroider-build/addon-blueprint/blob/main/files/.github/workflows/push-dist.yml) to bring back that capability.

Reviewer: ```suggestion
Note that we use features/syntax that require compilation, so you can't easily point package.json files at the git repo of your addon for easy cross-repo testing before it is published to NPM. There are [workflows you can use](https://github.com/embroider-build/addon-blueprint/blob/main/files/.github/workflows/push-dist.yml) to bring back that capability.
```
